forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,385 @@
+
+        let wallet_info = {
+            let wallet_address = self
+                .default_wallet_address
+                .clone()
+                .unwrap_or("address not set".to_string());
+
+            let wallet_balance = {
+                match balance(&self.default_wallet_address_balance) {
+                    Ok(bal) => format!("[balance: {}]", bal),
+                    Err(e) => e.to_string(),
+                }
+            };
+
+            format!(
+                "Default wallet address: {} [{}]",
+                wallet_address, wallet_balance
+            )
+        };
+
+        output.push(network);
+        output.push(uptime);
+        output.push(chain);
+        output.push(chain_health);
+        output.push(wallet_info);
+
+        output
+    }
+}
+
+impl InfoCommand {
+    pub async fn run(&self, config: Config, _opts: &CliOpts) -> anyhow::Result<()> {
+        let res = tokio::try_join!(
+            node_status((), &config.client.rpc_token),
+            chain_head(&config.client.rpc_token),
+            chain_get_name((), &config.client.rpc_token),
+            start_time(&config.client.rpc_token),
+            wallet_default_address((), &config.client.rpc_token)
+        );
+
+        match res {
+            Ok((node_status, head, network, start_time, default_wallet_address)) => {
+                let cur_duration: Duration = SystemTime::now().duration_since(UNIX_EPOCH)?;
+                let blocks_per_tipset_last_finality =
+                    node_status.chain_status.blocks_per_tipset_last_finality;
+
+                let default_wallet_address_balance = if let Some(def_addr) = &default_wallet_address
+                {
+                    let balance = wallet_balance((def_addr.clone(),), &config.client.rpc_token)
+                        .await
+                        .map_err(handle_rpc_err)?;
+                    Some(balance)
+                } else {
+                    None
+                };
+
+                let node_status_info = NodeStatusInfo::new(
+                    cur_duration,
+                    blocks_per_tipset_last_finality,
+                    head.0.as_ref(),
+                    start_time,
+                    network,
+                    default_wallet_address.clone(),
+                    default_wallet_address_balance,
+                );
+
+                for line in node_status_info.format(Utc::now()) {
+                    println!("{}", line);
+                }
+
+                Ok(())
+            }
+            Err(e) => Err(handle_rpc_err(e)),
+        }
+    }
+}
+
+fn balance(balance: &Option<String>) -> anyhow::Result<String> {
+    if let Some(bal) = balance {
+        let balance_token_amount = TokenAmount::from_atto(bal.parse::<BigInt>()?);
+        Ok(format!("{:.4}", balance_token_amount.pretty()))
+    } else {
+        Err(anyhow::anyhow!("could not find balance"))
+    }
+}
```

Comment:
It's very odd to have a function take in a single `&Option<T>` - just make the inner function, and call one of the `Option` combinators on the outer option

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8131fccabf844bba1255f586298b2ac7

Of course, in this example, the mapped function is trivial and should be inlined.